backends/mock_client.go

```diff
@@ -0,0 +1,52 @@
+package backends
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type MockReturnErrorBackend struct{}
```

Comment:
This doesn't use a mock framework, so please consider renaming it.

Reply:
Changed to `errorReturningBackend`.